### PR TITLE
Clean up type check codegen and param overrides

### PR DESCRIPTION
**Pull request by auscompgeek**
Also fix an off-by-one error in checking signed ints.

### CODE REVIEW

**Comment by virtuald** [comment]
Looks fine, but tests broken?

File: gen/hooks.py [hunk 1]
```diff
@@ -19,14 +19,31 @@
 # fmt: off
 
 def _gen_check(pname, ptype, strict=False):
+    SIGNED_CHECK = "isinstance({0}, int) and -1<<{1} <= {0} < 1<<{1}".format
```
Comment (virtuald) [inline gen/hooks.py]: /me hates `.format` ... but whatever I suppose.

Comment (auscompgeek) [reply gen/hooks.py]: Well, if it's the overhead you're worried about, none of this happens at runtime anyway! :stuck_out_tongue_winking_eye:

**Comment by virtuald** [comment]
:shipit:

**Pull request merged**